src/pages/OrderPage.tsx

```diff
@@ -167,20 +191,59 @@ const OrderPage = () => {
     reValidateMode: 'onChange',
   });
 
-  const onSubmit = (data: FormValues) => {
+  useEffect(() => {
+    if (user?.name) {
+      setValue('sender', user.name);
+    }
+  }, [user, setValue]);
+
+  const onSubmit = async (form: FormValues) => {
     if (!product) return;
 
+    if (recipients.length === 0) {
+      toast.error('받는 사람이 없습니다.');
+      return;
+    }
+
     alert(
       `주문이 완료되었습니다.\n` +
         `상품명: ${product.name}\n` +
         `구매 수량: ${recipients.reduce((sum, r) => sum + r.quantity, 0)}\n` +
-        `발신자 이름: ${data.sender}\n` +
-        `메시지: ${data.message}`
+        `발신자 이름: ${form.sender}\n` +
+        `메시지: ${form.message}`
     );
-    navigate(ROUTE.MAIN);
+
+    try {
+      const payload: OrderRequest = {
+        productId: product.id,
+        message: form.message,
+        messageCardId: String(selectedCardId),
+        ordererName: form.sender,
+        receivers: recipients.map((r) => ({
+          name: r.name,
+          phoneNumber: r.phone,
+          quantity: r.quantity,
+        })),
+      };
+
+      await postOrder(payload);
+      navigate(ROUTE.MAIN);
+    } catch (error: unknown) {
+      const err = error as AxiosError<{ data: { message: string } }>;
+      const status = err.response?.status;
+      const msg = err.response?.data?.data?.message || '주문에 실패했습니다.';
+
+      if (status === 401) {
```

Comment:
응답코드는 axios에 정의되어 있는데요. 수정이 필요한 부분은 아니지만 알아두시면 좋을 것 같습니다.
```suggestion
      if (status === HttpStatusCode.Unauthorized) {
```

Reply:
해당방식을 사용하면 유지보수에 좋을 것 같아 피드백 반영하겠습니다!